app/components/authorShowHeader/profileImageUpload.tsx

```diff
@@ -0,0 +1,118 @@
+import * as React from "react";
+import { Author, authorSchema } from "../../model/author/author";
+import { Dispatch, connect } from "react-redux";
+import { withStyles } from "../../helpers/withStylesHelper";
+import { updateProfileImage } from "../../actions/author";
+import { AppState } from "../../reducers";
+import { denormalize } from "normalizr";
+import { CurrentUser } from "../../model/currentUser";
+import CircularProgress from "@material-ui/core/CircularProgress";
+// import ArticleSpinner from "../common/spinner/articleSpinner";
+const styles = require("./profileImageUpload.scss");
+
+interface ProfileImageUploadProps {
+  author: Author;
+  currentUser: CurrentUser;
+  dispatch: Dispatch<any>;
+}
+
+interface ProfileImageUploadState {
+  imageUrl: string;
+  isLoading: boolean;
+}
+
+@withStyles<typeof ProfileImageUpload>(styles)
+class ProfileImageUpload extends React.PureComponent<ProfileImageUploadProps, ProfileImageUploadState> {
+  constructor(props: ProfileImageUploadProps) {
+    super(props);
+
+    this.state = {
```

Comment:
I think you don't need the component state because you can handle loading status and imageUrl string with Redux cycle. (with Reducer)